packages/mu-http/driver.js

```diff
@@ -14,4 +14,144 @@
+          sendTo.response.writeHead(200, { 'Content-Type': 'application/json' })
+          sendTo.response.write(stringify(message))
+          sendTo.response.end()
+        } else {
+          cb && cb(mue.transport('connection has alredy recieved response!'))
+        }
+      } else {
+
+        // client initiating connection to server
+        var body = []
+        var inbound
+        var req = http.request({host: options.target.host,
+                                port: options.target.port,
+                                method: 'POST',
+                                path: '/',
+                                headers: {'Content-Type': 'application/json'}})
+
+        req.on('response', function (response) {
+          response.on('data', function (chunk) {
+            body.push(chunk)
+          })
+
+          response.on('end', function () {
+            if (response.statusCode !== 200) {
+              receive(mue.transport(response.statusCode + ' ' + response.statusMessage), message)
+              return
+            }
+
+            inbound = parse(Buffer.concat(body))
+            receive(inbound.err, inbound.value)
+          })
+        })
+
+        req.on('error', function (err) {
+          cb(mue.transport(err))
+        })
+
+        req.write(stringify(message))
+        req.end()
+      }
+    }
+
+
+
+    function listen (port, host, ready) {
+      server = http.createServer()
+
+      server.on('request', function (request, response) {
+        var body = []
+        var inbound
+
+        request.on('data', function (chunk) {
+          body.push(chunk)
+        })
+
+        request.on('end', function () {
+          inbound = parse(Buffer.concat(body))
+
+          if (!connections[inbound.value.protocol.src]) {
+            connections[inbound.value.protocol.src] = []
+            connectionsByIp[request.socket.remoteAddress + '_' + request.socket.remotePort] = []
+          }
+          connections[inbound.value.protocol.src].push({request: request, response: response})
+          connectionsByIp[request.socket.remoteAddress + '_' + request.socket.remotePort] = inbound.value.protocol.src
+          receive(inbound.err, inbound.value)
+        })
+      })
+
+      return server.listen(port, host, ready)
+    }
+
+
+
+    function tearDown (cb) {
+      for (var conn in connections) {
+        if (connections[conn] && connections[conn].length > 0) {
+          var c = connections[conn].pop()
+          c.response.end()
+          c.request.destroy()
+        }
+      }
+      if (!server && cb) {
+        cb()
+        return
+      }
+      if (server) {
+        if (cb) {
+          setImmediate(function () {
+            server.close(cb)
+          })
+          return
+        }
+        server.close()
+      }
+    }
+
+
+
+    server = options.source && listen(options.source.port, options.source.host, options.ready)
+
+    if (!server && options.ready instanceof Function) {
+      options.ready()
```

Comment:
Memory is hazy on this, but there was definitely a good reason, I think it was needed for tests (or rather to avoid using setTimeouts in tests) - and may also be needed internally